src/consensus/read_validator.rs

```diff
@@ -98,8 +104,74 @@ impl ReadValidator {
         }
     }
 
+    fn get_validator_set(&self, height: u64) -> SnapchainValidatorSet {
+        let mut result = &self.validator_set[0];
+        for config in &self.validator_set {
+            if config.shard_ids.contains(&self.shard_id)
+                && config.effective_at <= height
+                && config.effective_at > result.effective_at
+            {
+                result = config;
+            }
+        }
+        result.validators.clone()
+    }
+
+    fn verify_signatures(&self, value: &proto::DecidedValue) -> bool {
+        let certificate = match value.value.as_ref().unwrap() {
+            proto::decided_value::Value::Shard(shard_chunk) => shard_chunk
+                .commits
+                .as_ref()
+                .unwrap()
+                .to_commit_certificate(),
+
+            proto::decided_value::Value::Block(block) => {
+                block.commits.as_ref().unwrap().to_commit_certificate()
+            }
+        };
+
+        let validator_set = self.get_validator_set(certificate.height.as_u64());
+
+        let mut expected_pubkeys = validator_set
+            .validators
+            .iter()
+            .map(|validator| validator.public_key.to_bytes());
+
+        if certificate.aggregated_signature.signatures.len() == 0
+            || certificate.aggregated_signature.signatures.len() != expected_pubkeys.len()
```

Comment:
It won't always be equal.  With tendermint you need at least 2/3rds of the validators to sign. I think you should use Malachite's `ThresholdParam::default().quorom.is_met(signatures_length, validators_length) `

https://github.com/informalsystems/malachite/blob/53a9d9e071e773ff959465f2836648d8ad2a5c74/code/crates/core-types/src/threshold.rs#L34

Reply:
My bad, fixed!